carrot-log4j-aws-sns/src/main/java/com/carrotgarden/log4j/aws/sns/Appender.java

```diff
@@ -203,8 +209,11 @@ public boolean hasAmazonClient() {
 
 	/** provide amazon login credentials from file */
 	protected boolean ensureCredentials() {
-
-        if (hasCredentials()) {
+		if (hasAwsCredentialsOverride()) {
+			amazonCredentials = new AWSCredentialsProviderChain(new InstanceProfileCredentialsProvider(),
+																new EnvironmentVariableCredentialsProvider(),
```

Comment:
Well, the difference here is that there's an actual file involved for part of the chain - of which might not exist if we're only using the override/system/IAM connection approach. Didn't want the properties provider to throw because it couldn't load the physical file that might not exist.